lms/djangoapps/course_api/blocks/views.py

```diff
@@ -309,6 +311,10 @@ def list(self, request, hide_access_denials=False):  # pylint: disable=arguments
         response = super().list(request, course_usage_key,
                                 hide_access_denials=hide_access_denials)
 
+        if POPULATE_USER_ACTIVITY_FLAG.is_enabled():
```

Comment:
i haven't actually tested some of the connection points with mobile locally, and there are a lot of calls, so i was being defensive here, because i didn't want one of these calls to trigger some exception before it had been thoroughly tested

Reply:
Yeah I didn't say get rid of the waffle. I was suggesting bringing the waffle check inside the method. So you only have one spot checking it.

Reply:
yup, i didn't think that's what you meant
i meant that i put the flag on the outside since i haven't tested some of the endpoints from the mobile app, so i was concerned that the function call itself could trigger an exception (rather than the code inside the function)

Reply:
I don't understand the concern - like you were worried you'd typo the function call? I think the linter would catch that. And you could still typo the waffle call to see if it was enabled.

Reply:
doesn't have to be a typo, could be that form.cleaned_data["course_id"] gives me a keyerror for example, although i guess if i was going to get import-related errors i'd still get them with the imports on the outside